tfhe/docs/getting_started/operations.md

```diff
@@ -1,64 +1,484 @@
-| `nor`          | Binary  |
-| `xnor`         | Binary  |
-| `cmux`         | Ternary |
+# Operations
 
-A walk-through using homomorphic Booleans can be found [here](../Boolean/tutorial.md).
+The structure and operations related to all types (i.e., Booleans, shortint and integer) are described in this section.
 
-## Shortint
+## Booleans
 
-In TFHE-rs, shortint represents short unsigned integers encoded over a maximum of 8 bits. A complete homomorphic arithmetic is provided, along with the possibility to compute univariate and bi-variate functions. Some operations are only available for integers up to 4 bits. More technical details can be found [here](../shortint/operations.md).
+Native homomorphic Booleans support common Boolean operations.
 
 The list of supported operations is:
 
-| Operation name            | Type         |
-| ------------------------- | ------------ |
-| Negation                  | Unary        |
-| Addition                  | Binary       |
-| Subtraction               | Binary       |
-| Multiplication            | Binary       |
-| Division\*                | Binary       |
-| Modular reduction         | Binary       |
-| Comparisons               | Binary       |
-| Left/Right Shift          | Binary       |
-| And                       | Binary       |
-| Or                        | Binary       |
-| Xor                       | Binary       |
-| Exact Function Evaluation | Unary/Binary |
-
-{% hint style="info" %}
-The division operation implements a subtlety: since data is encrypted, it might be possible to compute a division by 0. The division is tweaked so that dividing by 0 returns 0.
-{% endhint %}
-
-A walk-through example can be found [here](../shortint/tutorial.md), and more examples and explanations can be found [here](../shortint/operations.md).
+| name                                                          | symbol | type   |
+| ------------------------------------------------------------- | ------ | ------ |
+| [BitAnd](https://doc.rust-lang.org/std/ops/trait.BitAnd.html) | `&`    | Binary |
+| [BitOr](https://doc.rust-lang.org/std/ops/trait.BitOr.html)   | `\|`   | Binary |
+| [BitXor](https://doc.rust-lang.org/std/ops/trait.BitXor.html) | `^`    | Binary |
+| [Neg](https://doc.rust-lang.org/std/ops/trait.Neg.html)       | `!`    | Unary  |
+
+## ShortInt
+
+Native small homomorphic integer types (e.g., FheUint3 or FheUint4) easily compute various operations. In general, computing over encrypted data is as easy as computing over clear data, since the same operation symbol is used. The addition between two ciphertexts is done using the symbol `+` between two FheUint. Many operations can be computed between a clear value (i.e. a scalar) and a ciphertext.
+
+In Rust native types, any operation is modular. In Rust, `u8`, computations are done modulus 2^8. The similar idea is applied for FheUintX, where operations are done modulus 2^X. In the type FheUint3, operations are done modulo 8.
```

Comment:
In Rust, operations on native types are modular

Next sentence is just a repeat for u8 so maybe change to :

For example, computations on `u8` are done modulo `2^8`

Side note: I'm not sure operations are all modulor, for example some saturating ops (which arguably are special cases) are maybe not modular in the classic sense use for other ops

Reply:
I think, all operations tied to operators (+, -, *) are modular (they are `wrappin_` in release and `checked_op(rhs).unwrap()` in debug.
saturating ops are tied to special functions eg `saturatin_add`